pkg/operator/controller/extension/required/add.go

```diff
@@ -0,0 +1,163 @@
+	}
+	if r.Lock == nil {
+		r.Lock = &sync.RWMutex{}
+	}
+
+	r.clock = clock.RealClock{}
+
+	c, err := builder.
+		ControllerManagedBy(mgr).
+		Named(ControllerName).
+		For(&operatorv1alpha1.Extension{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: ptr.Deref(r.Config.Controllers.ExtensionRequired.ConcurrentSyncs, 0),
+		}).
+		Build(r)
+	if err != nil {
+		return err
+	}
+
+	for _, extension := range runtimeClusterExtensions {
+		eventHandler := mapper.EnqueueRequestsFrom(
+			ctx,
+			mgr.GetCache(),
+			r.MapObjectKindToExtensions(extension.objectKind, extension.newObjectListFunc),
+			mapper.UpdateWithNew,
+			c.GetLogger(),
+		)
+
+		// Execute the mapper function at least once to initialize the `KindToRequiredTypes` map.
+		// This is necessary for extension kinds which are registered but for which no extension objects exist in the
+		// garden runtime cluster (e.g. when backups are disabled).
+		// In such cases, no regular watch event is triggered, and the mapping function will not be executed.
+		// Thus, the extension kind would never be part of the `KindToRequiredTypes` map
+		// and the reconciler would not be able to decide whether the  Extension is required.
+		if err = c.Watch(&controllerutils.HandleOnce[client.Object, reconcile.Request]{Handler: eventHandler}); err != nil {
+			return err
+		}
+
+		if err := c.Watch(source.Kind[client.Object](mgr.GetCache(), extension.object, eventHandler, extensions.ObjectPredicate())); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// MapObjectKindToExtensions returns a mapper function for the given 'extensions.gardener.cloud' extension kind
+// that lists all existing resources of the given kind and stores the respective types in the `KindToRequiredTypes` map.
+// Afterwards, it returns all 'operator.gardener.cloud' Extensions that responsible for the given kind.
+func (r *Reconciler) MapObjectKindToExtensions(objectKind string, newObjectListFunc func() client.ObjectList) mapper.MapFunc {
+	return func(ctx context.Context, log logr.Logger, _ client.Reader, _ client.Object) []reconcile.Request {
+		log = log.WithValues("extensionKind", objectKind)
+
+		listObj := newObjectListFunc()
+		if err := r.Client.List(ctx, listObj); err != nil && !meta.IsNoMatchError(err) {
+			// Let's ignore bootstrap situations where extension CRDs were not yet applied. They will be deployed
+			// eventually by the garden controller.
+			log.Error(err, "Failed to list extension objects")
+			return nil
+		}
+
+		r.Lock.RLock()
+		oldRequiredTypes, kindCalculated := r.KindToRequiredTypes[objectKind]
+		r.Lock.RUnlock()
+		newRequiredTypes := sets.New[string]()
+
+		if err := meta.EachListItem(listObj, func(o runtime.Object) error {
+			obj, err := apiextensions.Accessor(o)
+			if err != nil {
+				return err
+			}
+
+			newRequiredTypes.Insert(obj.GetExtensionSpec().GetExtensionType())
```

Comment:
It should only react on resources with `ExtensionClassName` =`garden`

```suggestion
			if ptr.Deref(obj.GetExtensionSpec().GetExtensionClass(), "") == extensionsv1alpha1.ExtensionClassGarden {
				newRequiredTypes.Insert(obj.GetExtensionSpec().GetExtensionType())
			}			
```